src/System.Private.ServiceModel/src/System/ServiceModel/Channels/MessageContent.cs

```diff
@@ -180,15 +179,34 @@ protected override Task<Stream> CreateContentReadStreamAsync()
             // WriteMessageAsync might run synchronously and try to write to the stream. ProducerConsumerStream
             // will block on the write until the stream is being read from. The WriteMessageAsync method needs
             // to run on a different thread to prevent a deadlock.
-            _stream = new ProducerConsumerStream();
-            var bufferedStream = new BufferedWriteStream(_stream, WriteBufferSize);
-            Task.Run(() => _messageEncoder.WriteMessageAsync(_message, bufferedStream));
-            return Task.FromResult(_stream);
+            var resultStream = new ProducerConsumerStream();
```

Comment:
Just a couple questions to help me understand:
Before the change, _stream held the ProducerConsumerStream, and it looks like the BufferedWriteStream wrapped around it was just allowed to fall out of scope and be GC'ed after the write completed.

After the change, _stream holds the BufferedWriteStream, not the ProducerConsumerStream.

Questions are:
  1. Is this change intended to control the lifetime of the BufferedWriteStream and not let it be GC'ed till we're done?
  2. Are there any ramifications of _stream now holding the "outer" BufferedWriteStream instead of the "inner" ProducerConsumerStream?

Don't block on this question -- I just wanted to understand what's going on here.